cpp/include/utilities/cython.hpp

```diff
@@ -15,53 +15,113 @@
  */
 #pragma once
 
+#include <experimental/graph_view.hpp>
 #include <graph.hpp>
-// #include <experimental/graph_view.hpp>
 #include <raft/handle.hpp>
 
 namespace cugraph {
 namespace cython {
 
-// FIXME: use std::variant instead of a union if possible
-// FIXME: add both CSRView and graph_type_t objects for easier testing during
-//        the transition
-union graphUnion {
-  graphUnion() {}
-  GraphCSRView<int, int, float> GraphCSRViewFloat;
-  GraphCSRView<int, int, double> GraphCSRViewDouble;
-};
+enum class numberTypeEnum : int { intType, floatType, doubleType };
 
-enum weightTypeEnum { floatType = 0, doubleType = 1 };
+// FIXME: The GraphCSRView* types are not in use! Those are left in place in
+// case a legacy GraphCSRView class is needed, but these should be removed ASAP.
+enum class graphTypeEnum : int {
+  null,
+  GraphCSRViewFloat,
+  GraphCSRViewDouble,
+  graph_view_t_float,
+  graph_view_t_double,
+  graph_view_t_float_mg,
+  graph_view_t_double_mg,
+  graph_view_t_float_transposed,
+  graph_view_t_double_transposed,
+  graph_view_t_float_mg_transposed,
+  graph_view_t_double_mg_transposed
+};
 
-// FIXME: Add comments describing this struct, where it's used, etc.
+// "container" for a graph type instance which insulates the owner from the
+// specifics of the actual graph type. This is intended to be used in Cython
+// code that only needs to pass a graph object to another wrapped C++ API. This
+// greatly simplifies the Cython code since the Cython definition only needs to
+// define the container and not the various individual graph types in Cython.
 struct graph_container_t {
-  graph_container_t() {}
-  graphUnion graph;
-  // FIXME: cython issues using an enum so just using an int for now.
-  // weightTypeEnum wType;
-  int wType;
+  // FIXME: use std::variant (or a better alternative, ie. type erasure?) instead
+  //        of a union if possible
+  union graphPtrUnion {
+    void* null;
+    GraphCSRView<int, int, float>* GraphCSRViewFloatPtr;
+    GraphCSRView<int, int, double>* GraphCSRViewDoublePtr;
+    experimental::graph_view_t<int, int, float, false, false>* graph_view_t_float_ptr;
+    experimental::graph_view_t<int, int, double, false, false>* graph_view_t_double_ptr;
+    experimental::graph_view_t<int, int, float, false, true>* graph_view_t_float_mg_ptr;
+    experimental::graph_view_t<int, int, double, false, true>* graph_view_t_double_mg_ptr;
+    experimental::graph_view_t<int, int, float, true, false>* graph_view_t_float_transposed_ptr;
+    experimental::graph_view_t<int, int, double, true, false>* graph_view_t_double_transposed_ptr;
+    experimental::graph_view_t<int, int, float, true, true>* graph_view_t_float_mg_transposed_ptr;
+    experimental::graph_view_t<int, int, double, true, true>* graph_view_t_double_mg_transposed_ptr;
+  };
+
+  inline graph_container_t() : graph_ptr_union{nullptr}, graph_ptr_type{graphTypeEnum::null} {}
+
+  inline ~graph_container_t()
```

Comment:
The destructor deletes the pointer within the union.  A reasonable thing to do.

Probably need to decide the desired life cycle behavior of this class.  If the intention is to create the graph container at the cython level and pass it by reference (probably const reference since I don't think you would want a caller to change anything) then we should delete the copy constructor, assignment operators, etc so that we won't have pointer issues.

Reply:
Agreed. I added a copy ctor and assignment operator and deleted them.  I also added a comment in the `graph_container_t` struct describing the typical usage/lifespan of the struct, but moving forward if we need to use this differently we can support and document that as well.